rust/core-lib/src/plugins/manifest.rs

```diff
@@ -107,10 +111,132 @@ pub enum PluginScope {
     SingleInvocation,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+/// Represents a custom command provided by a plugin.
+pub struct Command {
+    /// Human readable title, for display in (for example) a menu.
+    pub title: String,
+    /// A short description of the command.
+    pub description: String,
+    /// Template of the command RPC as it should be sent to the plugin.
+    pub rpc_cmd: PlaceholderRpc,
+    /// A list of `CommandArgument`s, which the client should use to build the RPC.
+    pub args: Vec<CommandArgument>,
+}
+
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+/// A user provided argument to a plugin command.
+pub struct CommandArgument {
+    /// A human readable name for this argument, for use as placeholder
+    /// text or equivelant.
+    pub title: String,
+    /// A short (single sentence) description of this argument's use.
+    pub description: String,
+    pub key: String,
+    pub arg_type: ArgumentType,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    /// If `arg_type` is `Choice`, `options` must contain a list of options.
+    pub options: Option<Vec<ArgumentOption>>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub enum ArgumentType {
+    Number, Int, PosInt, Bool, String, Choice,
```

Comment:
Correct, but I don't think it's a big deal, especially since it conveys that the type can be more complex, and I care about languages with good type systems like Rust and Swift having a nice type-safe way of representing our protocol.